odoo/tools/mail.py

```diff
@@ -265,7 +265,7 @@ def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=Fals
 # ----------------------------------------------------------
 
 URL_REGEX = r'(\bhref=[\'"](?!mailto:|tel:|sms:)([^\'"]+)[\'"])'
-TEXT_URL_REGEX = r'https?://[a-zA-Z0-9@:%._\+~#=/-]+(?:\?\S+)?'
+TEXT_URL_REGEX = r'https?://[\w@:%.+&~#=/-]+(?:\?\S+)?'
```

Comment:
it's not immediately obvious why you've removed `_` from the regex? It's not valid inside domain names, but it is valid in the URL path, which we seem to be supporting here? 🤔

Reply:
It's still included in `\w`. From https://docs.python.org/3/library/re.html
> For Unicode (str) patterns:
Matches Unicode word characters; this includes most characters that can be part of a word in any language, as well as numbers and the underscore. If the [ASCII](https://docs.python.org/3/library/re.html#re.ASCII) flag is used, only [a-zA-Z0-9_] is matched.